Source/UZKArchive.m

```diff
@@ -1973,6 +1999,11 @@ - (BOOL)performWriteAction:(int(^)(uLong *crc, NSError * __autoreleasing*innerEr
         UZKLogDebug("Making zip_fileinfo struct for date %{time_t}ld", lrint(fileDate.timeIntervalSince1970));
         zip_fileinfo zi = [UZKArchive zipFileInfoForDate:fileDate];
         
+        if (posixPermissions > 0) {
+            
+            zi.external_fa = (posixPermissions + 32768) << 16;
+        }
+        
```

Comment:
Perhaps it would make sense to put an else clause here, and assign `zi.external_fa` based on the file's actual permissions if nothing is specified. But if we do that, it changes behavior, and I would wait until after v1.9 for that. I would bump the version number to v2.0 so people would know to check. What do you think?

Reply:
I think it's okay, keep the actual permissions of the file without specifying content.